edb/schema/pointers.py

```diff
@@ -1001,6 +1006,67 @@ def _deparse_name(
             return ref
 
 
+class PointerAlterFragment(
+    referencing.ReferencedObjectCommandBase[Pointer]
+):
+    @classmethod
+    def _cmd_tree_from_ast(
+        cls,
+        schema: s_schema.Schema,
+        astnode: qlast.DDLOperation,
+        context: sd.CommandContext,
+    ) -> referencing.AlterReferencedInheritingObject[Any]:
+        cmd = super()._cmd_tree_from_ast(schema, astnode, context)
+        assert isinstance(cmd, PointerCommand)
+        if isinstance(astnode, qlast.CreateConcreteLink):
+            cmd._process_create_or_alter_ast(schema, astnode, context)
+        else:
+            expr_cmd = qlast.get_ddl_field_command(astnode, 'expr')
+            if expr_cmd is not None:
+                expr = expr_cmd.value
+                if expr is None:
+                    # `DROP EXPRESSION` detected
+                    aop = sd.AlterObjectProperty(
+                        property='expr',
+                        new_value=s_expr.Expression(text='DROP EXPRESSION'),
```

Comment:
What's the reason for the `DROP EXPRESSION` hack? Seems like setting it to `None` and then using `get_attribute_set_cmd()` in `canonicalize_attributes` would be cleaner.

Reply:
I guess that works (I just realized that's almost what `get_local_attribute_value` is doing behind the scenes).